lib/patches/db/pg.rb

```diff
@@ -39,6 +39,20 @@ class PG::Connection
     alias_method :exec_params_without_profiling, :exec_params
   end
 
+  def prepare_parameterized_statement(args)
+    if args[0].match(/\$[1-9]/).present? && args[1].is_a?(Array)
```

Comment:
.match can be expensive here on every sql query, maybe check for the array first? is there any need to run a regex if you have an array and the array has a length > 0 

```
if Array === args[1] && args[1].length > 0
```

Reply:
Refreshed commit with your advice
[](https://github.com/MiniProfiler/rack-mini-profiler/pull/405/commits/584553ea27f72b80572e00962c0130d714e8ae3a)